### PR TITLE
Remove assert statement from non-test files

### PR DESCRIPTION
**Pull request by harishsg993010**

### CODE REVIEW

**Comment by gbaned** [comment]
@harishsg993010 Can you please sign CLA. Thank you!

**Comment by gbaned** [comment]
@harishsg993010 Any update on this PR? Please. Thank you!

File: tfx/components/trainer/rewriting/rewriter_test.py [hunk 1]
```diff
@@ -54,20 +54,24 @@ def name(self):
       return 'my_rewriter'
 
     def _pre_rewrite_validate(self, original_model):
-      assert original_model == self._expected_original_model
+      if original_model != self._expected_original_model:
```
Comment (chongkong) [inline tfx/components/trainer/rewriting/rewriter_test.py:57]: This is a test file so assert is okay

File: tfx/components/transform/executor.py [hunk 2]
```diff
@@ -167,32 +167,38 @@ def __init__(self,
 
   @property
   def file_pattern(self):
-    assert self._file_pattern
+    if not self._file_pattern:
```
Comment (chongkong) [inline tfx/components/transform/executor.py:170]: Please avoid using empty error message (fill with some message.)

File: tfx/components/transform/executor.py [hunk 3]
```diff
@@ -469,8 +480,9 @@ def Do(self, input_dict: Dict[str, List[types.Artifact]],
     for split in splits_config.analyze:
       data_uris = artifact_utils.get_split_uris(
           input_dict[standard_component_specs.EXAMPLES_KEY], split)
-      assert len(data_uris) == len(
-          examples_file_formats), 'Length of file formats is different'
+      if len(data_uris) != len(
+          examples_file_formats):
```
Comment (chongkong) [inline tfx/components/transform/executor.py:484]: please fix line break

File: tfx/components/transform/executor.py [hunk 4]
```diff
@@ -480,8 +492,9 @@ def Do(self, input_dict: Dict[str, List[types.Artifact]],
     for split in splits_config.transform:
       data_uris = artifact_utils.get_split_uris(
           input_dict[standard_component_specs.EXAMPLES_KEY], split)
-      assert len(data_uris) == len(
-          examples_file_formats), 'Length of file formats is different'
+      if len(data_uris) != len(
```
Comment (chongkong) [inline tfx/components/transform/executor.py:495]: please fix line break

File: tfx/components/transform/executor.py [hunk 5]
```diff
@@ -1716,8 +1732,9 @@ def _AssertSameTFXIOSchema(self, datasets: Sequence[_Dataset]) -> None:
     if not datasets:
       return
     for dataset in datasets[1:]:
-      assert (datasets[0].tfxio.ArrowSchema().equals(
-          dataset.tfxio.ArrowSchema()))
+      if not (datasets[0].tfxio.ArrowSchema().equals(
```
Comment (chongkong) [inline tfx/components/transform/executor.py:1735]: unnecessary parenthesis

File: tfx/components/transform/executor_test.py [hunk 6]
```diff
@@ -253,7 +253,8 @@ def _verify_transform_outputs(self,
     if materialize:
       expected_outputs.append('transformed_examples')
 
-      assert len(example_artifacts) == len(transformed_example_artifacts)
+      if len(example_artifacts) != len(transformed_example_artifacts):
+        raise AssertionError
```
Comment (chongkong) [inline tfx/components/transform/executor_test.py:257]: this is a test file so assert should be okay. please revert.

File: tfx/types/value_artifact_test.py [hunk 7]
```diff
@@ -33,7 +33,8 @@ class _MyValueArtifact(value_artifact.ValueArtifact):
   TYPE_NAME = 'MyValueTypeName'
 
   def encode(self, value: str):
-    assert isinstance(value, str), value
+    if not isinstance(value, str):
+      raise AssertionError(value)
```
Comment (chongkong) [inline tfx/types/value_artifact_test.py:37]: this is a test file so should be okay. please revert

File: tfx/utils/typing_utils.py [hunk 8]
```diff
@@ -68,13 +68,15 @@ def is_compatible(value: Any, tp: Any) -> bool:
...
+      if not maybe_args:
+        raise AssertionError(f'{tp} should be subscripted.')
       return any(is_compatible(value, arg) for arg in maybe_args)
     # Type[T]
     elif maybe_origin is type:
       if not maybe_args:
         return inspect.isclass(value)
-      assert len(maybe_args) == 1
+      if len(maybe_args) != 1:
+        raise AssertionError
```
Comment (chongkong) [inline tfx/utils/typing_utils.py:79]: please give a descriptive error message. same for the below.

**Review by chongkong — CHANGES_REQUESTED** [review]
Sorry I was reviewing this PR old time ago but I had a 500 error at Github (and some of my comments not saved 😂) at the time and then didn't come back for a while.

**Comment by gbaned** [comment]
@harishsg993010 Can you please sign CLA. Thank you!

**Comment by gbaned** [comment]
Hi @harishsg993010 Any update on this PR? Please. Thank you!

**Pull request closed without merging**